### PR TITLE
Support Bootstrap v4 alpha 4 in v1

### PR DESCRIPTION
**Pull request by justin808**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/bootstrap-loader/170)

### CODE REVIEW

**Comment by justin808** [comment]
@rmobis @Andre-Gl can you please verify this works with webpack v1? Thanks.

**Comment by justin808** [comment]
@Judahmeek, could you maybe try this out?

**Comment by Judahmeek** [comment]
Yes, I'll try it out.

File: src/utils/createBootstrapRequire.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function(module, bootstrapVersion, bootstrapPath) {
   const scriptsPath = (
     parseInt(bootstrapVersion, 10) === 3 ?
     ['assets', 'javascripts', 'bootstrap'] :
-    ['dist', 'js', 'umd']
+    ['js', 'dist']
```
Comment (frickt) [inline src/utils/createBootstrapRequire.js]: add new path also to readme jquery module loader.

File: README.md [hunk 2]
```diff
@@ -15,7 +16,7 @@ Successor to [bootstrap-sass-loader](https://github.com/shakacode/bootstrap-sass
 2. We'd be thrilled to have another maintainer join us to help with Bootstrap 4 issues.
 3. We'd also be thrilled if any companies are open to sponsoring the development of features and issues regarding Bootstrap 4.
  
-That being said, Bootstrap 4 probably works just fine!
+That being said, Bootstrap 4 probably works just fine! You must use `4.0.0-alpha.4` or greater.
```
Comment (outdooricon) [inline README.md]: I believe line 40 also needs to be updated

**Comment by outdooricon** [comment]
We're really looking forward to seeing this in. Alpha2 is broken in multiple ways for us that alpha4 fixes

**Comment by Judahmeek** [comment]
@justin808 Ran it at https://ide.c9.io/judahmeek/bootstrap-loader
```
judahmeek:~/workspace/examples/basic (support-alpha-4-in-v1) $ npm run bs4:default:prod

> bootstrap-loader-example@1.0.0 bs4:default:prod /home/ubuntu/workspace/examples/basic
> npm run bs4:default:setup && npm run build && npm run prod


> bootstrap-loader-example@1.0.0 bs4:default:setup /home/ubuntu/workspace/examples/basic
> cp -f .bootstraprc-4-example .bootstraprc


> bootstrap-loader-example@1.0.0 build /home/ubuntu/workspace/examples/basic
> npm run clean && webpack --config webpack.prod.config.js


> bootstrap-loader-example@1.0.0 clean /home/ubuntu/workspace/examples/basic
> rm -rf public/

=> bootstrap-loader configuration: bootstrap-loader/extractStyles
Hash: a32c32e63843ea847c41
Version: webpack 1.13.2
Time: 18647ms
                               Asset     Size  Chunks             Chunk Names
ecb4572a5e478b107dfcb60c16a7eefa.ttf   107 kB          [emitted]  
25a32416abee198dd821b0b17a198a8f.eot  76.5 kB          [emitted]  
1dc35d25e61d819a9c357074014867ab.ttf   153 kB          [emitted]  
d7c639084f684d66a1bc66855d193ed8.svg   392 kB          [emitted]  
                              app.js   595 kB       0  [emitted]  main
                             app.css   559 kB       0  [emitted]  main
   [0] multi main 64 bytes {0} [built]
  [17] ./app/scripts/app.js 128 bytes {0} [built]
    + 17 hidden modules

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/alert in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 1:0-42

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/button in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 2:0-43

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/carousel in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 3:0-45

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/collapse in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 4:0-45

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/dropdown in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 5:0-45

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/modal in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 6:0-42

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/popover in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 7:0-44

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/scrollspy in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 8:0-46

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/tab in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 9:0-40

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/tooltip in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 10:0-44

ERROR in ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js
Module not found: Error: Cannot resolve 'file' or 'directory' ../bootstrap/dist/js/umd/util in /home/ubuntu/workspace/examples/basic/node_modules/bootstrap-loader
 @ ./~/bootstrap-loader/lib/bootstrap.scripts.loader.js!./~/bootstrap-loader/no-op.js 11:0-41
Child extract-text-webpack-plugin:
                                   Asset    Size  Chunks             Chunk Names
    ecb4572a5e478b107dfcb60c16a7eefa.ttf  107 kB          [emitted]  
        + 3 hidden modules

> bootstrap-loader-example@1.0.0 prod /home/ubuntu/workspace/examples/basic
> nodemon --watch app/markup --ext html server.prod.js

[nodemon] 1.10.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: /home/ubuntu/workspace/examples/basic/app/markup/**/*
[nodemon] starting `node server.prod.js`
=> 🚀  Production server is running on port 4000
^C
judahmeek:~/workspace/examples/basic (support-alpha-4-in-v1) $ git status
On branch support-alpha-4-in-v1
Your branch is up-to-date with 'origin/support-alpha-4-in-v1'.
nothing to commit, working tree clean
```

**Comment by justin808** [comment]
@Judahmeek @outdooricon I can't finish this for at least a week. Can one of you guys submit a new PR with this, fixing the things that are broken. 

thanks!

**Comment by Judahmeek** [comment]
@justin808 I'll give it a shot.

File: src/utils/createBootstrapRequire.js (hunk 1 above)
Comment (Judahmeek) [inline src/utils/createBootstrapRequire.js]: What is this readme jquery module loader you speak of?

**Comment by Judahmeek** [comment]
@justin808 All the errors were due to the fact that the examples have their own modules of bootstrap-loader instead of being linked to local code. Is there a particular reason for that or can I attempt to create a local dependency with symlinks to the local source code?

Also, the examples' webpack.configs need to be updated on line 56/67 (just removing the `/umd\`).

**Comment by justin808** [comment]
@Judahmeek 

https://github.com/shakacode/bootstrap-loader/tree/v1/examples/basic

See the line

`npm run install-local`

We can hard code the values in the file, but they need to be changed every release.

Maybe we just need to change the docs? on this page.

**Comment by Judahmeek** [comment]
We can probably just put this down to my inexperience as a contributor. I overlooked the README, went straight to package.json, and overlooked `npm install-local` there because I wasn't considered the need for the examples to be used as integration tests (and thus the need for some form of linking).

**Comment by igama** [comment]
Hey, Any update on this ? :)

**Comment by justin808** [comment]
I just merged this to the V1 branch as is. There might be a couple more changes and then a release for v1.

---

Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/bootstrap-loader/170#-:-KSV-xmPDNWW7MZdyf6m:b-2b0hfe)*

**Comment by justin808** [comment]
Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/utils/createBootstrapRequire.js, line 15 at r2](https://reviewable.io:443/reviews/shakacode/bootstrap-loader/170#-KRH5FLIe52CRx_rol0K-r2-15:-KSV0hPBTJB56xD5d1x0:bjaxg5f) ([raw file](https://github.com/shakacode/bootstrap-loader/blob/f9bbc26d39e602124fc75190b19a408f8ebac5af/src/utils/createBootstrapRequire.js#L15)):*
<details><summary><i>Previously, Judahmeek (Judah Meek) wrote…</i></summary>
> What is this readme jquery module loader you speak of?
</details>
@judahmeek 

I'll update this.

```
module: {
  loaders: [
    // Use one of these to serve jQuery for Bootstrap scripts:

    // Bootstrap 3
    { test:/bootstrap-sass[\/\\]assets[\/\\]javascripts[\/\\]/, loader: 'imports?jQuery=jquery' },

    // Bootstrap 4
    { test: /bootstrap[\/\\]dist[\/\\]js[\/\\]umd[\/\\]/, loader: 'imports?jQuery=jquery' },
  ],
},

```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/bootstrap-loader/170)*

**Comment by justin808** [comment]
@outdooricon @frickt @igama @rmobis @Andre-Gl:

Released 1.2 with support for alpha.v4!

**Comment by outdooricon** [comment]
@justin808 sweet, thanks!

**Pull request merged**